compiler/rustc_typeck/src/check/upvar.rs

```diff
@@ -624,6 +632,17 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         false
     }
 
+    /// Returns a Ty that applies the specified capture kind on the provided Ty
+    fn apply_capture_kind_on_ty(&self, ty: Ty<'tcx>, capture_kind: UpvarCapture<'tcx>) -> Ty<'tcx> {
```

Comment:
Hmm, so `ty` doen't have any associations with capture kind on their own. I'd rename this to be `apply_capture_kind_on_capture_ty`.

Also it feels like a helper so instead of ti being a method on FnCtx I'd just make it a free function that just accepts a `tcx: TyCtxt<'tcx>`